### PR TITLE
Accuracy validation doesn't depend on batch size and number of GPU

### PR DESCRIPTION
**Pull request by ljaljushkin**
Previously, last data point in all DataLoaders were dropped to prevent issue with BN (#379)
But it's actual for training only, on eval it shouldn't happen.

### CODE REVIEW

**Comment by ljaljushkin** [comment]
sota eval 414 - WIP

**Comment by ljaljushkin** [comment]
> sota eval 414 - WIP

eval is OK, except of binary models, but there're fixed in 415

**Review by vshampor — APPROVED** [review]
